### PR TITLE
ENH: latex styler bool pandas options

### PR DESCRIPTION
**Pull request by attack68**
- [x] adds the last  remaining options for #41395 for the existing args in Styler.to_latex.

### CODE REVIEW

**Comment by jreback** [comment]
can we add only the most important options here, e.g. it is perfectly fine to have keyword args that are not set by a global option. the namespace is getting cluttered.

**Comment by attack68** [comment]
sure, do you want to review some of the other ones that went in, and, to be consistent in terms of importance, remove some others also? 

Always easier to add something back than to remove it after a release.

**Comment by jreback** [comment]
> sure, do you want to review some of the other ones that went in, and, to be consistent in terms of importance, remove some others also?
> 
> Always easier to add something back than to remove it after a release.

yeah i think let's keep the options to the current exposed ones in prior versions e.g. don't add non-super-common options (i know vague)

**Comment by attack68** [comment]
> > sure, do you want to review some of the other ones that went in, and, to be consistent in terms of importance, remove some others also?
> > Always easier to add something back than to remove it after a release.
> 
> yeah i think let's keep the options to the current exposed ones in prior versions e.g. don't add non-super-common options (i know vague)

I removed ``css_convert`` and ``siunitx`` since they are probably more rarely used, but ``hrules`` I think should stick becuase DataFrame.to_html and Styler.to_html, have different defaults, the DataFrame version can only use True and Styler can render both with a False default.

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -502,9 +502,10 @@ def to_latex(
             \\<position_float>
 
             Cannot be used if ``environment`` is "longtable".
-        hrules : bool, default False
+        hrules : bool
             Set to `True` to add \\toprule, \\midrule and \\bottomrule from the
             {booktabs} LaTeX package.
+            Defaults to ``pandas.options.styler.latex.hrules``, which is `False`.
```
Comment (jreback) [inline pandas/io/formats/style.py:508]: can you add a versionchanged 1.4 here

File: pandas/io/formats/style.py [hunk 2]
```diff
@@ -515,16 +516,17 @@ def to_latex(
         sparse_index : bool, optional
             Whether to sparsify the display of a hierarchical index. Setting to False
             will display each explicit level element in a hierarchical key for each row.
-            Defaults to ``pandas.options.styler.sparse.index`` value.
+            Defaults to ``pandas.options.styler.sparse.index``, which is `True`.
         sparse_columns : bool, optional
             Whether to sparsify the display of a hierarchical index. Setting to False
             will display each explicit level element in a hierarchical key for each
-            column. Defaults to ``pandas.options.styler.sparse.columns`` value.
+            column. Defaults to ``pandas.options.styler.sparse.columns``, which
```
Comment (jreback) [inline pandas/io/formats/style.py:523]: same

**Review by jreback — CHANGES_REQUESTED** [review]
looks fine, doc comment, ping on green.

File: pandas/io/formats/style.py (hunk 2 above)
Comment (attack68) [reply pandas/io/formats/style.py:523]: this one hasn't changed, just clarified docs.

File: pandas/io/formats/style.py (hunk 1 above)
Comment (attack68) [reply pandas/io/formats/style.py:508]: done

**Comment by attack68** [comment]
@jreback ping green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
cool

**Pull request merged**